### PR TITLE
[FIX]: Tackle Old PRs

### PR DESCRIPTION
**Pull request by achamma723**
This PR tackles PRs from the old repo:
- Add dcrt with unit tests and example

### CODE REVIEW

File: examples/plot_dcrt_example.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    rho = 0.8
+    # Nominal false positive rate
+    alpha = 5e-2
+
+    X, y, _, __ = multivariate_1D_simulation(
+        n_samples=n, n_features=p, support_size=n_signal, rho=rho, seed=sim_ind
+    )
+
+    # Applying a reLu function on the outcome y to get non-linear relationships
+    y = np.array([max(0.0, i) for i in y])
```
Comment (bthirion) [inline examples/plot_dcrt_example.py:38]:
```suggestion
    y = np.max(0.0, y)
```

File: hidimstat/dcrt.py [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+    solver="liblinear",
+    fdr_control="bhq",
+    n_jobs=1,
+    verbose=False,
+    joblib_verbose=0,
+    ntree=100,
+    type_prob="regression",
+    random_state=2022,
+):
+
```
Comment (bthirion) [inline hidimstat/dcrt.py:43]: deserves  a docstring...

File: hidimstat/dcrt.py [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+    if verbose:
+        return selected, pvals, ts
+
+    return selected
+
+
+def _x_distillation_lasso(
+    X, idx, Sigma_X=None, cv=3, n_regus=100, alpha=None, use_cv=False, n_jobs=1
+):
+
```
Comment (bthirion) [inline hidimstat/dcrt.py:163]: A one-line docstring is welcome

**Review by bthirion — COMMENTED** [review]
Thx ! 
Still needs more tests.

**Comment by achamma723** [comment]
@bthirion, this PR is ready to review and merge

File: hidimstat/dcrt.py [hunk 4]
```diff
@@ -0,0 +1,420 @@
...
+    eps_res = y - X_minus_idx.dot(coef_minus_idx)
+    sigma2_y = np.mean(eps_res**2)
+
+    # T follows Gaussian distribution
+    ts = np.dot(eps_res, X_res) / np.sqrt(n_samples * sigma2_X * sigma2_y)
+
+    return ts
+
+
+# def _optimal_reg_param(
```
Comment (bthirion) [inline hidimstat/dcrt.py:338]: Can you remove that ?

**Review by bthirion — COMMENTED** [review]
There is room for improvement, but I prefer to merge it as-is and improve later.
Can you remove the unused code, then we merge ?
Thx in advance,

File: hidimstat/dcrt.py [hunk 5]
```diff
@@ -0,0 +1,388 @@
...
+        raise ValueError(f"{statistic} statistic is not supported.")
+    ts = np.zeros(n_features)
+    ts[selection_set] = np.array([i for i in results])
+
+    if scaled_statistics:
+        ts = (ts - np.mean(ts)) / np.std(ts)
+
+    if statistic in ["residual", "randomforest"]:
+        pvals = np.minimum(2 * stats.norm.sf(np.abs(ts)), 1)
+    # elif statistic == "likelihood":
```
Comment (bthirion) [inline hidimstat/dcrt.py:229]: Please remove such lines

**Review by bthirion — COMMENTED** [review]
Still a very minor thing and we merge

**Comment by achamma723** [comment]
Hello @bthirion, I think it is ready for the merge now

**Review by bthirion — APPROVED** [review]
LGTM, thx.

**Comment by bthirion** [comment]
Merging now.

**Pull request merged**